runhouse/resources/secrets/secret.py

```diff
@@ -217,18 +217,23 @@ def extract_provider_secrets(cls, names: List[str] = None) -> Dict[str, "Secret"
 
     # TODO: refactor this code to reuse rns_client save_config code instead of rewriting
     def save(
-        self, name: str = None, save_values: bool = True, headers: Optional[Dict] = None
+        self,
+        name: str = None,
+        save_values: bool = True,
+        headers: Optional[Dict] = None,
+        folder: str = None,
```

Comment:
This shouldn't only be a feature of Secrets, any time a subresource is saved we want to save it in the same folder as the parent. I think we should add folder as an arg to _save_sub_resources.